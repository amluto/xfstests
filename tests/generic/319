#! /bin/bash
# FS QA Test No. 001
#
# Test cmtime update after writes to shared writable mmaps
#
#-----------------------------------------------------------------------
# Copyright (c) 2013 Andrew Lutomirski.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
    rm -f "$TEST_DIR/file"
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter

# Modify as appropriate.
_supported_fs generic
_supported_os Linux

"$here/src/mmap_cmtime" "$TEST_DIR/file" 2>&1 | tee "$seqres.full" |grep -v '^  \[DEBUG\]'
status=${PIPESTATUS[0]}

exit $status
