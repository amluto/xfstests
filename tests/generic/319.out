QA output created by 319

test_basic
Write page 0
munmap

test_write_twice
Write page 0
Write page 0
munmap

test_write_twice_sync(0)
Write page 0
Write page 0
fsync
munmap

test_write_twice_sync(1)
Write page 0
Write page 0
MS_SYNC one byte
munmap

test_write_twice_sync(2)
Write page 0
Write page 0
MS_SYNC entire mapping byte
munmap

test_write_twice_sync(3)
Write page 0
Write page 0
MS_ASYNC one byte
munmap

test_write_twice_sync(4)
Write page 0
Write page 0
MS_ASYNC entire mapping byte
munmap

test_write_twice_sync(5)
Write page 0
Write page 0
fdatasync
munmap

test_write_twice_sync(6)
Write page 0
Write page 0
sync_file_range one byte
munmap

test_two_pages
Write page 0
Write page 1
Write page 1
munmap

Passed!
